### PR TITLE
[infoware] Add port of infoware library

### PR DESCRIPTION
**Pull request by ThePhD**
This pull request adds a port for the [C++ infoware library](https://github.com/ThePhD/infoware).

All triplets should be supported, with both dynamic and static builds available.

I could not find additional examples of CI that needed to change or more. The version of infoware being targeted by vcpkg should be 0.5.3.

### CODE REVIEW

File: ports/infoware/CONTROL [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# vcpkg CONTROL file for infoware
```
Comment (PhoebeHui) [inline ports/infoware/CONTROL:1]: Could you please remove the comments?

File: ports/infoware/CONTROL [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# vcpkg CONTROL file for infoware
+Source: infoware
+Version: v0.5.3
+Description: C++ Library for pulling system and hardware information, without hitting the command line.
```
Comment (PhoebeHui) [inline ports/infoware/CONTROL:4]: Could you also add ' Homepage' field?

File: ports/infoware/CONTROL [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+Description: Prefer usage of X11 to find graphical capabilities.
+
+Feature: d3d
+Description: Prefer usage of Direct3D to find graphical capabilities (typically only works on Windows systems).
+
+Feature: opencl
+Description: Prefer usage of OpenCL to find graphical capabilities of the system.
+
+Feature: opengl
+Description: Prefer usage of OpenGL to find graphical capabilities (may require additional libraries to be available for linking depending on the system).
```
Comment (PhoebeHui) [inline ports/infoware/CONTROL:16]: Does these feature test pass locally? The CI does not test feature currently.

File: ports/infoware/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,44 @@
+# vcpkg portfile.cmake file for infoware
+
+include(vcpkg_common_functions)
```
Comment (PhoebeHui) [inline ports/infoware/portfile.cmake:3]: Could you please remove it? this is deprecated.

File: ports/infoware/portfile.cmake [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        -DINFOWARE_EXAMPLES=OFF
+        -DINFOWARE_TESTS=OFF
+        -DGIT_EXECUTABLE=${GIT}
+        -DGIT_FOUND=true
+)
+
+vcpkg_install_cmake()
+
+# make sure the LICENSE file is present as the copyright file
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/infoware RENAME copyright)
```
Comment (PhoebeHui) [inline ports/infoware/portfile.cmake:39]: We perfer to using 'file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT}  RENAME copyright)' instead.

**Review by PhoebeHui — CHANGES_REQUESTED** [review]
Could you also update the hash value?

The CI test failures:
  File does not have expected hash:

          File path: [ C:/agent/_work/1/s/downloads/temp/ThePhD-infoware-v0.5.3.tar.gz ]
      Expected hash: [ 64a4f0b1027a20bf9c90bf99d638947cb1c0d6bad1b22c8b0403713c5e2c69756435906996b154989f9eaa41e5d55743d7bf5ef51f7bbf8d99eae94d29af1b2c ]
        Actual hash: [ 217bb9332214d823445e19f2c465199536e89a36faccea8cb72f8dd41a177e1739969d131ad25d1878e688a3a6cc1290c2125ef9d2205ad4fd334b24b27d491a ]

  The file may have been corrupted in transit.  This can be caused by
  proxies.  If you use a proxy, please set the HTTPS_PROXY and HTTP_PROXY
  environment variables to
  "https://user:password@your-proxy-ip-address:port/".

File: ports/infoware/CONTROL [hunk 6]
```diff
@@ -0,0 +1,16 @@
+# vcpkg CONTROL file for infoware
+Source: infoware
+Version: v0.5.3
```
Comment (PhoebeHui) [inline ports/infoware/CONTROL:3]: Could you update the Version to '0.5.3'?

**Review by PhoebeHui — CHANGES_REQUESTED** [review]

File: ports/infoware/CONTROL (hunk 3 above)
Comment (ThePhD) [reply ports/infoware/CONTROL:16]: We test OpenGL and OpenCL locally, but D3D and X11 (the defaults) are CI tested.

**Comment by ThePhD** [comment]
> Files changes (133)

I, uh. Seem to have irreversibly ruined my git commit. One moment. ;;

**Comment by PhoebeHui** [comment]
@ThePhD, the CI test with following triplet, does it expect? could you please take a look?

infoware:x64-uwp
infoware:arm-uwp
infoware:arm64-windows

**Comment by ThePhD** [comment]
I was not expecting this: there seems to be a specific problem with MSBuild on these platforms and building the tool to generate PCI data? Works fine with Ninja; I'll try to fix it up for MSBuild. Sorry about the failed tests...!

**Comment by ThePhD** [comment]
I actually don't understand what's going on in vcpkg in these cases. I can build for the platforms listed as failures independently outside of vcpkg, invoking the CMake build and install with either Ninja or MSBuild just fine. But inside of vcpkg, I get errors like these:

> Desktop/vcpkg/buildtrees/infoware/arm64-windows-dbg/pciids/pci.ids > infoware_generated/pci_data.hpp"
FAILED: infoware_generated/pci_data.hpp 
cmd.exe /C "cd /D Desktop\vcpkg\buildtrees\infoware\arm64-windows-dbg && Desktop\vcpkg\downloads\tools\cmake-3.17.2-windows\cmake-3.17.2-win32-x86\bin\cmake.exe -E make_directory infoware_generated/ && Desktop\vcpkg\buildtrees\infoware\arm64-windows-dbg\infoware_pci_generator.exe Desktop/vcpkg/buildtrees/infoware/arm64-windows-dbg/pciids/pci.ids > infoware_generated/pci_data.hpp"
This version of Desktop\vcpkg\buildtrees\infoware\arm64-windows-dbg\infoware_pci_generator.exe is not compatible with the version of Windows you're running. Check your computer's system information and then contact the software publisher.
ninja: build stopped: subcommand failed.

I sincerely am not sure how to handle this. Maybe I should just disable the `x64_uwp`, `arm_uwp`, and `arm64` builds?

**Comment by strega-nil** [comment]
@ThePhD For UWP, I have no idea what's wrong; I would just set it as `fail`. There's often an issue getting ports which aren't designed for it to work on UWP.

For `arm64-windows`, the issue that you have is that you have build tools that need to run on the host architecture, but you're always running them on the target architecture; it's completely valid to do that, but it means cross-compilation is impossible. You may want to pull a boost, and require that the host binaries are installed before building the target libraries, or you may want to just wait until we have support for tool-depends. If the latter, I would set it to fail and write a comment to the effect of `This port doesn't support cross-compilation`.

**Comment by ThePhD** [comment]
Is the right thing to do `Supports: !(uwp|arm)` or is it better to modify `scripts/ci.baseline.txt`?

**Comment by ThePhD** [comment]
Also, I think the x64_osx pipeline is broken...!

**Comment by PhoebeHui** [comment]
Currently we add the follow code when it unsupport some arch or platform.

1. CONTROL: Supports: !(uwp|arm)
2. Portfile.cmake:  vcpkg_fail_port_install(ON_ARCH "arm"  ON_TARGET  "uwp")
3. ci.baseline.txt: 
inforware:arm-uwp=fail
inforware:arm64-windows=fail
inforware:x64-uwp=fail

Let's wait for the CI test results.

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
@ThePhD Thanks for your contribution!

**Comment by ThePhD** [comment]
Oh, nice, thanks for adding that function to the PR, glad it seems to all work now!!

**Review by PhoebeHui — APPROVED** [review]

**Comment by dan-shaw** [comment]
Thanks for the PR!

**Pull request merged**